Controller/OrderController.cs

```diff
@@ -0,0 +1,481 @@
+using Models;
+using Models.Contract;
+using ECommerce.Models;
+using ECommerce.Services;
+using System.Linq.Expressions;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using CustomExceptions;
+using ECommerce.Repositories;
+using Constants;
+
+namespace ECommerce.Controllers
+{
+	public class OrderController
+	{
+		private readonly IOrderService _orderService;
+		private readonly IUserService _userService;
+		private readonly IProductService _productService;
+
+		public OrderController(IOrderService orderService, IUserService userService, IProductService productService)
+		{
+			_orderService = orderService;
+			_userService = userService;
+			_productService = productService;
+
+			_orderService.OnOrderProcessed += HandleOrderProcessed;
+		}
+
+		public void PlaceOrder(Guid userId)
+		{
+			try
+			{
+				Customer user = (Customer)_userService.GetUserById(userId);
+
+				if (user.Role != UserRole.Customer)
+				{
+					Console.WriteLine("Only customers can place orders.");
+					return;
+				}
+				List<Product> products = _productService.GetAllProducts().ToList();
+
+				DisplayAvailableProducts(products);
+
+				var productSerialNumbers = new List<int>();
+				List<Guid> productIds = new List<Guid>();
+				decimal totalAmount = 0;
+
+				while (true)
+				{
+					Console.WriteLine("Enter a choice");
+					Console.WriteLine("1. Add product");
+					Console.WriteLine("2. Finish order");
+
+					(var choice, bool wantToGoBack) = GetChoiceInput();
+					if (wantToGoBack)
+					{
+						Console.Clear();
+						return;
+					}
+
+
+					if (choice == Choice_Constants.ONE)
```

Comment:
Insetad 'One' you can make it more readable, e.g. AddProduct.
You can also use enums for this